File: typescript/base/src/main/vsdk/toolkit/material/XxdfComponentFlag.ts
export class XxdfComponentFlag {
  public static readonly DIFFUSE_COMPONENT = 1;
  public static readonly GLOSSY_COMPONENT = 2;
  public static readonly SPECULAR_COMPONENT = 4;

  private constructor() {
  }
}

export class XxdfComponentFlagInfo {
  public static readonly XXDF_COMPONENTS = 3;
  public static readonly NO_COMPONENTS = 0;
  public static readonly ALL_COMPONENTS = XxdfComponentFlag.DIFFUSE_COMPONENT
    | XxdfComponentFlag.GLOSSY_COMPONENT
    | XxdfComponentFlag.SPECULAR_COMPONENT;

  private constructor() {
  }
}

File: java/base/src/main/java/vsdk/toolkit/material/XxdfComponentFlag.java
package vsdk.toolkit.material;

public final class XxdfComponentFlag {
    public static final int DIFFUSE_COMPONENT = 1;
    public static final int GLOSSY_COMPONENT = 2;
    public static final int SPECULAR_COMPONENT = 4;

    private XxdfComponentFlag() {
    }
}

final class XxdfComponentFlagInfo {
    static final int XXDF_COMPONENTS = 3;
    static final int NO_COMPONENTS = 0;
    static final int ALL_COMPONENTS = XxdfComponentFlag.DIFFUSE_COMPONENT
        | XxdfComponentFlag.GLOSSY_COMPONENT
        | XxdfComponentFlag.SPECULAR_COMPONENT;

    private XxdfComponentFlagInfo() {
    }
}
